pkg/agent/nodeportlocal/portcache/port_table_windows_test.go

```diff
@@ -0,0 +1,152 @@
+//go:build windows
+// +build windows
+
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package portcache
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	portcachetesting "antrea.io/antrea/pkg/agent/nodeportlocal/portcache/testing"
+	"antrea.io/antrea/pkg/agent/nodeportlocal/rules"
+	rulestesting "antrea.io/antrea/pkg/agent/nodeportlocal/rules/testing"
+)
+
+func TestRestoreRules(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	allNPLPorts := []rules.PodNodePort{
+		{
+			NodePort: nodePort1,
+			PodPort:  1001,
+			PodIP:    podIP,
+			Protocol: "tcp",
+		},
+		{
+			NodePort: nodePort1,
+			PodPort:  1001,
+			PodIP:    podIP,
+			Protocol: "udp",
+		},
+		{
+			NodePort: nodePort2,
+			PodPort:  1002,
+			PodIP:    podIP,
+			Protocol: "udp",
+		},
+	}
+
+	mockPortRules.EXPECT().AddRule(nodePort1, podIP, 1001, "tcp")
+	mockPortRules.EXPECT().AddRule(nodePort1, podIP, 1001, "udp")
+	mockPortRules.EXPECT().AddRule(nodePort2, podIP, 1002, "udp")
+
+	syncedCh := make(chan struct{})
+	err := portTable.RestoreRules(allNPLPorts, syncedCh)
+	require.NoError(t, err)
+}
+
+func TestDeleteRule(t *testing.T) {
+	var err error
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	npData := &NodePortData{
+		NodePort: startPort,
+		PodIP:    podIP,
+		PodPort:  1001,
+		Protocol: ProtocolSocketData{
+			Protocol: "tcp",
+		},
+	}
+
+	tcs := []struct {
+		name       string
+		testForPod bool
+	}{
+		{
+			name:       "Delete Rule",
+			testForPod: true,
+		},
+		{
+			name:       "Delete Rules For Pod",
+			testForPod: false,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			portTable.addPortTableCache(npData)
+			mockPortRules.EXPECT().DeleteRule(startPort, podIP, 1001, "tcp")
+			if tc.testForPod {
+				err = portTable.DeleteRule(podIP, 1001, "tcp")
+			} else {
+				err = portTable.DeleteRulesForPod(podIP)
+			}
```

Comment:
1. When testForPod is true, it tests DeleteRule, which looks confusing.
2. It's actually not common to test two different functions in one test, orginazing them as table driven style, which is usually about expected input and output for same function. To share code for multiple tests, extracting common code to a shared function is more common.
3. However, I think the point of `DeleteRulesForPod` is it can delete all rules for the specific pod IP, unlike `DeleteRule` can only delete one specified rule. So an appropriate test for the former is to construct multiple rules first, then make sure a single call can clean up all of them.